File: Projects/leetcode/CreateTargetArray.java
package leetcode;

import java.util.*;

class CreateTargetArray {
    public int[] createTargetArray(int[] nums, int[] index) {
        List<Integer> res = new ArrayList<>();
        for (int i = 0; i < nums.length; i++) {
            res.add(index[i], nums[i]);
        }
        return res.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: Projects/leetcode/CreateTargetArray.kt
package leetcode

import java.util.ArrayList
import java.util.function.ToIntFunction

internal class CreateTargetArray {
    fun createTargetArray(nums: IntArray, index: IntArray): IntArray {
        val res: List<Int> = ArrayList()
        for (i in nums.indices) {
            res.add(index[i], nums[i])
        }
        return res.stream().mapToInt { obj: Int -> obj }.toArray()
    }
}